### PR TITLE
Move DeduplicationPeriod to ledger-api-domain [KVL-1047]

### PR DESCRIPTION
**Pull request by nicu-da**
- To enable use the reuse of `DeduplicationPeriod`, so it can be passed down from the API to the submitter info
- I'm not at all convinced about this move, please feel free to propose another option

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [x] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by SamirTalwar-DA — APPROVED** [review]
Seems sensible to me.

**Review by fabiotudone-da — COMMENTED** [review]
Agree with @SamirTalwar-DA, perhaps let's wait for a feedback from @digital-asset/kv-participant just to make sure there aren't caveats we are overlooking.

**Review by tudor-da — APPROVED** [review]
LGTM, although I'd appreciate @rautenrieth-da 's feedback on this one as well.

File: ledger/ledger-api-domain/src/main/scala/com/digitalasset/ledger/api/DeduplicationPeriod.scala
Comment (tudor-da) [inline ledger/ledger-api-domain/src/main/scala/com/digitalasset/ledger/api/DeduplicationPeriod.scala:4]: Where is this going to be used as part of the API. If it is only as part of command submission, I am wondering whether this shouldn't be in `com.daml.ledger.api.messages.command.completion`. l

Comment (nicu-da) [reply ledger/ledger-api-domain/src/main/scala/com/digitalasset/ledger/api/DeduplicationPeriod.scala:4]: Yes, so in the API it's gonna be part of `com.daml.ledger.api.Commands`. i'm up for moving it anywhere as I am not really familiar with the structure we have here

**Review by miklos-da — APPROVED** [review]
Looks good to me, i.e., deduplication period is not only relevant to participant state but is a more general concept used throughout the stack (both by the participant and the committer).

File: ledger/ledger-api-domain/src/main/scala/com/digitalasset/ledger/api/DeduplicationPeriod.scala
Comment (rautenrieth-da) [reply ledger/ledger-api-domain/src/main/scala/com/digitalasset/ledger/api/DeduplicationPeriod.scala:4]: `com.daml.ledger.api.messages.*` seems to only contain types corresponding to top-level messages used in the ledger API.

**Review by rautenrieth-da — APPROVED** [review]
The move looks sensible to me.

**Pull request merged**
